torch/csrc/jit/ir.cpp

```diff
@@ -566,6 +568,135 @@ Value* Value::setUniqueName(const std::string & name) {
   return this;
 }
 
+template<typename T>
```

Comment:
This functionality is duplicated with the changes I am working on to add actual primitive types, so we will have to sort this out when it merges.  The functionality seems basically equivalent though so it should not be possible to do.